### PR TITLE
OSSM-3745 Refactor TestEgressGateways (T13)

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/tests/tasks/traffic/egress/egress_gateways_test.go [hunk 1]
```diff
@@ -16,109 +16,267 @@ package egress
...
+			t.LogStep("Create a ServiceEntry to external istio.io")
+			oc.ApplyString(t, ns, ExServiceEntry)
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, ExServiceEntry)
+			})
+			assertExternalHTTPRequestSuccessful(t, ns)
+
+			t.LogStep("Create a Gateway to external istio.io")
+			oc.ApplyTemplate(t, ns, ExGatewayTemplate, smcp)
+			t.Cleanup(func() {
```
Comment (fjglira) [inline pkg/tests/tasks/traffic/egress/egress_gateways_test.go:53]: A question only, if you create two cleanup func are going to be executed in sequence when the test case finishes or is aborted by failure?

**Review by fjglira — APPROVED** [review]
lgtm

File: pkg/tests/tasks/traffic/egress/egress_gateways_test.go (hunk 1 above)
Comment (luksa) [reply pkg/tests/tasks/traffic/egress/egress_gateways_test.go:53]: All cleanup functions get executed (in reverse order).

Comment (fjglira) [reply pkg/tests/tasks/traffic/egress/egress_gateways_test.go:53]: Ohhh, good to know that.

**Pull request merged**